x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/use_cti_event_counts.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState, useMemo } from 'react';
+import { usePrevious } from 'react-use';
+import { i18n } from '@kbn/i18n';
+import { MatrixHistogramType } from '../../../../common/search_strategy/security_solution';
+import { useKibana } from '../../../common/lib/kibana';
+import { ThreatIntelLinkPanelProps } from '../../components/overview_cti_links';
+import { convertToBuildEsQuery } from '../../../common/lib/keury';
+import { esQuery } from '../../../../../../../src/plugins/data/public';
+import { useMatrixHistogram } from '../../../common/containers/matrix_histogram';
+import { EVENT_DATASET } from '../../../../common/cti/constants';
+import { TimeRange } from '../../../resolver/types';
+
+export const ID = 'ctiEventCountQuery';
+const PREFIX = 'threatintel.';
+
+export const useCTIEventCounts = (
+  { deleteQuery, filters, from, indexNames, indexPattern, setQuery, to }: ThreatIntelLinkPanelProps,
+  timeRange: TimeRange
+) => {
+  const [isInitialLoading, setIsInitialLoading] = useState(true);
+  const { uiSettings } = useKibana().services;
+
+  useEffect(() => {
+    return () => {
+      if (deleteQuery) {
+        deleteQuery({ id: ID });
+      }
+    };
+  }, [deleteQuery]);
+
+  const matrixHistogramRequest = useMemo(
+    () => ({
+      endDate: to,
+      errorMessage: i18n.translate('xpack.securitySolution.overview.errorFetchingEvents', {
+        defaultMessage: 'Error fetching events',
+      }),
+      filterQuery: convertToBuildEsQuery({
+        config: esQuery.getEsQueryConfig(uiSettings),
+        indexPattern,
+        queries: [{ query: `event.kind = "enrichment"`, language: 'kql' }],
```

Comment:
A few notes here:

1. Is this actually KQL? I would expect the syntax to be `event.kind: "enrichment"`
2. I don't see this filter reflected in the network request, and I don't think it's being applied, so we're receiving ALL events in these indices, and then filtering down in memory to those whose datasets start with `threatintel`.
3. We're doing a lot of unnecessary work with this query:
    * Using an interval of 12h is unnecessary and just costs extra time/bandwidth/summing logic on the client; we just want total counts over the entire interval
    * We're also returning `hits` but not using those; you can specify `size: 0` on the outer query to prevent these from being returned.

Reply:
Also: I think `event.type: "indicator"` is more accurate than `event.kind: "enrichment"`; I frankly don't know what the latter is trying to convey and I'd suggest we focus on the former.